vm/vm.go

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2018 De-labtory
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package vm
+
+import (
+	"github.com/DE-labtory/koa/opcode"
+	"github.com/pkg/errors"
+)
+
+var opcodes = map[opcode.Type]Opcode{
+	opcode.Add:  Add{},
+	opcode.Push: Push{},
+}
+
+type Item uint32
+
+func Execute(instructions []Item) (Stack, error) {
+	stack := NewStack()
```

Comment:
I think second reason seems to be problem of naming. The reason why i implement `InstructionBuffer` is to prevent direct access to instructions.
so it seems to require additional functions such as moving the pointer for operations such as a jump.

Reply:
If the purpose of `InstructionBuffer` is to prevent direct access to instructions, then how about create `InstructionBuffer` concrete implementations **outside `Execute` function** using `instructions` then inject just `InstructionBuffer` to `Execute` parameters??

Reply:
Sounds good. but vm will check some validation in `Execute` so `Execute` needs `instructions`. validation checking function is added...! sorry :)